app/team.go

```diff
@@ -160,6 +161,124 @@ func (a *App) SoftDeleteAllTeamsExcept(teamID string) *model.AppError {
 	return nil
 }
 
+func (a *App) shouldCreateOnboardingLinkedBoard(c request.CTX, teamId string) bool {
+	ffEnabled := a.Config().FeatureFlags.OnboardingAutoShowLinkedBoard
+	if !ffEnabled {
+		return false
+	}
+
+	hasBoard, err := a.HasBoardProduct()
+	if err != nil {
+		a.Log().Error("error checking the existence of boards product: ", mlog.Err(err))
+		return false
+	}
+
+	if !hasBoard {
+		a.Log().Warn("board product not found")
+		return false
+	}
+
+	const preferenceName = "linked_board_created"
+	data, sysValErr := a.Srv().Store().System().GetByName(model.PreferenceOnboarding + "_" + preferenceName)
+	if sysValErr != nil {
+		var nfErr *store.ErrNotFound
+		if errors.As(sysValErr, &nfErr) { // if no board has been registered, it can create one for this team
+			return true
+		}
+		a.Log().Error("cannot get the system values", mlog.Err(sysValErr))
+		return false
+	}
+
+	// get the team list and check if the team value has been already stored, if so, no need to create a board in town square in that team
+	teamsList := strings.Split(data.Value, ",")
+	for _, team := range teamsList {
+		if team == teamId {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (a *App) createOnboardingLinkedBoard(c request.CTX, teamId string) (*fb_model.Board, *model.AppError) {
+	const defaultChannelName = "town-square"
+	const defaultTemplatesTeam = "0"
+	const defaultTemplateTitle = "Welcome to Boards!"
+	const preferenceName = "linked_board_created"
+	userId := c.Session().UserId
+
+	boardServiceItf, ok := a.Srv().services[product.BoardsKey]
+	if !ok {
+		return nil, model.NewAppError("CreateBoard", "app.team.create_onboarding_linked_board.product_key_not_found", nil, "", http.StatusBadRequest)
+	}
+
+	boardService, typeOk := boardServiceItf.(product.BoardsService)
+	if !typeOk {
+		// boardServiceItf is NOT of type product.BoardsService
+		return nil, model.NewAppError("CreateBoard", "app.team.create_onboarding_linked_board.itf_not_of_type", nil, "", http.StatusBadRequest)
+	}
+
+	templates, err := boardService.GetTemplates(defaultTemplatesTeam, userId)
+	if err != nil {
+		return nil, model.NewAppError("CreateBoard", "app.team.create_onboarding_linked_board.error_getting_templates", nil, "", http.StatusBadRequest).Wrap(err)
+	}
+
+	channel, appErr := a.GetChannelByName(c, defaultChannelName, teamId, false)
+	if appErr != nil {
+		return nil, appErr
+	}
+
+	var template *fb_model.Board = nil
+	for _, t := range templates {
+		v := t.Title
+		if v == defaultTemplateTitle {
```

Comment:
I would recommend using the same "id" I used in work templates to select a board, if the title of the board change (or if they get localized eventually), your code will break. The ID I target will not change in time: https://github.com/mattermost/mattermost-server/blob/master/app/work_template_executor.go#L178-L188
cc @wiggin77 to correct if I'm wrong.

Reply:
@JulienTant do you mean use the trackingTemplateId value from the Welcome to Boards template? that seems like a generated id and I am not sure if we can trust that. I would rather to keep the current approach since I think the name won't change in the forseable future and below I have a fallback that in case the template is not found, always use the templates in the 0 position, which if I understand correctly will always be the Welcome to boards template. @JulienTant @wiggin77 thoughts?

Reply:
Yeah I mean use the tracking ID. That's what been recommended to me by Doug as the best way to target a board.

Reply:
@pvev, @JulienTant the `trackingTemplateId` is designed specifically to represent a template in an way that does not change from release to release.  The title text may change but the trackingTemplateId will not.  Also we are looking at possibly supporting multilingual card titles so matching the title text may not work long term. 

Another thing to be aware of is that all code should be resilient to the absence of any particular template.  It is not difficult for admins to remove them.

Reply:
ok, understood. Thanks @wiggin77